packages/sit-onyx/src/components/OnyxRadioButton/OnyxRadioButton.vue

```diff
@@ -8,34 +9,128 @@ export type RadioButtonProps<TValue> = SelectionProps<TValue> & {
    * See also: https://html.spec.whatwg.org/multipage/input.html#radio-button-group
    */
   name: string;
+  isRequired?: boolean;
+  errorMessage?: string;
 };
 
 const props = defineProps<RadioButtonProps<TValue>>();
+
+const selector = ref<HTMLInputElement>();
+
+watch(
+  [() => props.errorMessage, selector],
+  () => selector.value?.setCustomValidity(props.errorMessage ?? ""),
+  { immediate: true },
+);
```

Comment:
```suggestion
watchEffect(() => {
  selector.value?.setCustomValidity(props.errorMessage ?? "");
});
```

Reply:
I tend to prefer `watch` over `watchEffect` as it can have drastic performance impacts. But for a one-liner like this it's actually a great idea!